api/app/controllers/spree/api/v2/platform/resource_controller.rb

```diff
@@ -18,6 +18,8 @@ class ResourceController < ::Spree::Api::V2::ResourceController
           def create
             resource = model_class.new(permitted_resource_params)
 
+            ensure_current_store(resource)
```

Comment:
@MatthewKennedy let's also add this behavior as a test to https://github.com/spree/spree/blob/master/api/spec/requests/spree/api/v2/platform/resources_spec.rb

Reply:
Will do.